optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/optional/score/MeetingSchedulingConstraintProvider.java

```diff
@@ -86,8 +89,12 @@ protected Constraint requiredAttendanceConflict(ConstraintFactory constraintFact
                         equal((leftRequiredAttendance, rightRequiredAttendance, leftAssignment) -> rightRequiredAttendance
                                 .getMeeting(),
                                 MeetingAssignment::getMeeting),
-                        filtering((leftRequiredAttendance, rightRequiredAttendance, leftAssignment,
-                                rightAssignment) -> rightAssignment.calculateOverlap(leftAssignment) > 0))
+                        overlaps((attendee1, attendee2, assignment) -> assignment.getStartingTimeGrain().getGrainIndex(),
+                                (attendee1, attendee2, assignment) -> assignment.getStartingTimeGrain().getGrainIndex() +
+                                        assignment.getMeeting().getDurationInGrains(),
+                                assignment -> assignment.getStartingTimeGrain().getGrainIndex(),
+                                assignment -> assignment.getStartingTimeGrain().getGrainIndex() +
+                                        assignment.getMeeting().getDurationInGrains()))
                 .penalizeConfigurable("Required attendance conflict",
                         (leftRequiredAttendance, rightRequiredAttendance, leftAssignment, rightAssignment) -> rightAssignment
                                 .calculateOverlap(leftAssignment));
```

Comment:
(not a change request)
This brings up an interesting question: often the overlap is penalized. For example in conference scheduling, the overlap of 2 somehow conflicting talks is penalized.
We must have the overlap/duration API to allow for indexing. Should we also provide a helper method to calculate the overlap during penalization? Probably we can't: LocalDate and Integer don't have a common interface with a "subtract" method.

Reply:
If there was such an interface, it would have to allow the following operations: `LocalDate - LocalDate -> Period`, `Integer - Integer -> Integer`, `LocalDateTime - LocalDateTime -> Duration`, etc. `overlapping` can be done for any `Comparable` class, `overlappingAmount` cannot; it requires an additional interface `difference(a,b)`. No such interface exists that all implement, although each class has their subtract method.

What we can do is help something like this:
`Pair<T,T> overlappingRange(interval1, interval2)`
which returns (max(A.start, B.start), min(A.end, B.end)) (which is the "area" of overlap). Then they can do the subtract operation on those two points (or can provide a function that that turns it into a number) See https://stackoverflow.com/a/36035369